charts/opentelemetry-collector/templates/hpa.yaml

```diff
@@ -9,7 +9,12 @@ metadata:
 spec:
   scaleTargetRef:
     apiVersion: apps/v1
+  {{- if and .Values.autoscaling.enabled (eq .Values.mode "deployment") }}
     kind: Deployment
+  {{- end }}
+  {{- if and .Values.autoscaling.enabled (eq .Values.mode "statefulset") }}
+    kind: StatefulSet
+  {{- end }}
```

Comment:
I believe we can simplify this to something like:

```
kind {{ title  .Values.mode }}
```

Reply:
updated